pages/[username].tsx

```diff
@@ -0,0 +1,198 @@
+import Image from "next/image";
+import LayoutContainer from "@components/LayoutContainer";
+import RedirectUtils from "utils/redirect";
+import Toggle from 'react-toggle'
+import UserAgentUtils from "utils/userAgent";
+import getConfig from "next/config";
+import parser from 'ua-parser-js'
+import useToastHook from "@hooks/toast/useToastHook"
+import React, { useEffect, useRef, useState } from "react";
+import { BaseContainer } from "@components/Page/BaseContainer";
+import { GetServerSidePropsContext } from "next";
+import { Header } from "@components/Header";
+import { Helmet } from "react-helmet";
+import { LoaderWrapper } from "@components/LoaderWrapper";
+import { MessageEnum } from "@shared/enum";
+import { toast } from "react-toastify";
+import { useGetProfile } from "@services/profile/profileHooks";
+import { useRouter } from "next/router";
+
+interface PageProps {
+    username?: string
+    isDynamicLink?: boolean
+}
+
+const { publicRuntimeConfig } = getConfig()
+
+export default function Profile(props: PageProps) {
+    const {isDynamicLink, username} = props
+    const { data, isLoading } = useGetProfile(username);
+    const [message, setMessage] = useState('')
+    const router = useRouter();
+    useEffect(() => {
+        if (data) {
+            localStorage.setItem(MessageEnum.targetUser, data.user_id);
+        }
+    }, [data])
+
+    useEffect(() => {
+        if (isDynamicLink) router.push(`/${username}`, undefined, { shallow: true })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isDynamicLink])
+
+    const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+    useEffect(() => {
+        adjustTextareaHeight();
+    }, [message]);
+
+    const adjustTextareaHeight = () => {
+        if (textAreaRef.current) {
+            const lineHeight = parseFloat(getComputedStyle(textAreaRef.current).lineHeight);
+            textAreaRef.current.style.height = `${lineHeight}px`;
+            const lines = Math.floor(textAreaRef.current.scrollHeight / lineHeight);
+            const numLines = lines <= 3 ? lines : 3;
+            textAreaRef.current.style.height = `${numLines * lineHeight}px`;
+        }
+    };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `adjustTextareaHeight` is defined inside a `useEffect` hook, which is not a good practice as it can lead to unnecessary re-renders. It would be better to define this function outside of the `useEffect` hook and then call it within the hook.
```diff
    useEffect(() => {
        adjustTextareaHeight();
    }, [message]);

    const adjustTextareaHeight = () => {
        if (textAreaRef.current) {
            const lineHeight = parseFloat(getComputedStyle(textAreaRef.current).lineHeight);
            textAreaRef.current.style.height = `${lineHeight}px`;
            const lines = Math.floor(textAreaRef.current.scrollHeight / lineHeight);
            const numLines = lines <= 3 ? lines : 3;
            textAreaRef.current.style.height = `${numLines * lineHeight}px`;
        }
    };
+    const adjustTextareaHeight = () => {
+        if (textAreaRef.current) {
+            const lineHeight = parseFloat(getComputedStyle(textAreaRef.current).lineHeight);
+            textAreaRef.current.style.height = `${lineHeight}px`;
+            const lines = Math.floor(textAreaRef.current.scrollHeight / lineHeight);
+            const numLines = lines <= 3 ? lines : 3;
+            textAreaRef.current.style.height = `${numLines * lineHeight}px`;
+        }
+    };
+
+    useEffect(() => {
+        adjustTextareaHeight();
+    }, [message]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->